src/openrct2-ui/windows/Finances.cpp

```diff
@@ -341,1048 +260,820 @@ static constexpr const uint32_t WindowFinancesPageHoldDownWidgets[] =
+            }
+        }
     }
-    else
+
+    void OnDrawResearch(rct_drawpixelinfo& dpi)
     {
-        w->width = WW_OTHER_TABS;
-        w->height = WH_OTHER_TABS;
+        WindowResearchFundingPagePaint(this, &dpi, WIDX_RESEARCH_FUNDING);
     }
-    window_event_resize_call(w);
-    window_event_invalidate_call(w);
-
-    WindowInitScrollWidgets(w);
-    w->Invalidate();
 
-    // Scroll summary all the way to the right, initially.
-    if (w->page == WINDOW_FINANCES_PAGE_SUMMARY)
-        WindowFinancesSummaryInvertscroll(w);
-}
+#pragma endregion
 
-static void WindowFinancesSetPressedTab(rct_window* w)
-{
-    int32_t i;
-    for (i = 0; i < WINDOW_FINANCES_PAGE_COUNT; i++)
-        w->pressed_widgets &= ~(1ULL << (WIDX_TAB_1 + i));
-    w->pressed_widgets |= 1LL << (WIDX_TAB_1 + w->page);
-}
+    void InitializeScrollPosition(rct_widgetindex widgetIndex, int32_t scrollId)
+    {
+        const auto& widget = this->widgets[widgetIndex];
+        scrolls[scrollId].h_left = std::max(0, scrolls[scrollId].h_right - (widget.width() - 2));
 
-static void WindowFinancesDrawTabImage(rct_drawpixelinfo* dpi, rct_window* w, int32_t page, int32_t spriteIndex)
-{
-    rct_widgetindex widgetIndex = WIDX_TAB_1 + page;
+        WidgetScrollUpdateThumbs(this, widgetIndex);
+    }
 
-    if (!(w->disabled_widgets & (1LL << widgetIndex)))
+    void DrawTabImage(rct_drawpixelinfo& dpi, int32_t tabPage, int32_t spriteIndex)
     {
-        if (w->page == page)
+        rct_widgetindex widgetIndex = WIDX_TAB_1 + tabPage;
+
+        if (!IsWidgetDisabled(widgetIndex))
         {
-            int32_t frame = w->frame_no / 2;
-            if (page == WINDOW_FINANCES_PAGE_SUMMARY)
-                frame %= 8;
-            spriteIndex += frame;
+            if (this->page == tabPage)
+            {
+                int32_t frame = frame_no / 2;
+                spriteIndex += (frame % window_finances_tab_animation_frames[this->page]);
+            }
+
+            gfx_draw_sprite(
+                &dpi, ImageId(spriteIndex), windowPos + ScreenCoordsXY{ widgets[widgetIndex].left, widgets[widgetIndex].top });
         }
+    }
 
-        gfx_draw_sprite(
-            dpi, ImageId(spriteIndex),
-            w->windowPos + ScreenCoordsXY{ w->widgets[widgetIndex].left, w->widgets[widgetIndex].top });
+    void DrawTabImages(rct_drawpixelinfo& dpi)
+    {
+        DrawTabImage(dpi, WINDOW_FINANCES_PAGE_SUMMARY, SPR_TAB_FINANCES_SUMMARY_0);
+        DrawTabImage(dpi, WINDOW_FINANCES_PAGE_FINANCIAL_GRAPH, SPR_TAB_FINANCES_FINANCIAL_GRAPH_0);
+        DrawTabImage(dpi, WINDOW_FINANCES_PAGE_VALUE_GRAPH, SPR_TAB_FINANCES_VALUE_GRAPH_0);
+        DrawTabImage(dpi, WINDOW_FINANCES_PAGE_PROFIT_GRAPH, SPR_TAB_FINANCES_PROFIT_GRAPH_0);
+        DrawTabImage(dpi, WINDOW_FINANCES_PAGE_MARKETING, SPR_TAB_FINANCES_MARKETING_0);
+        DrawTabImage(dpi, WINDOW_FINANCES_PAGE_RESEARCH, SPR_TAB_FINANCES_RESEARCH_0);
     }
+};
+
+static FinancesWindow* FocusOrCreateFinancesWindow()
```

Comment:
I don't really see the point of this function. Is this the style from another window?